client/src/pages/ScanStore/ScanStore.tsx

```diff
@@ -166,12 +170,16 @@ function ScanStore() {
   };
 
   const dismissRemoveConfirmation = () => {
-    setRemoveBarcode("");
     setShowRemoveConfirmation(false);
   };
 
   const raiseRemoveConfirmation = (barcode: string) => {
-    setRemoveBarcode(barcode);
+    // Find the item information for what is to be deleted
+    const itemToRemove = cartItems
+      .map((cartItem) => cartItem.item)
+      .find((item) => item.barcode === barcode);
+    if (!itemToRemove) return; // Should not be the case here
```

Comment:
If this would never happen, can just ignore

Reply:
during runtime it shouldn't be the case, but static compile-time check raises error for the next line, hence this check. Updated comment to explain the case here

Reply:
I see.